doc/users/next_whats_new/imshow_extent_units.rst

```diff
@@ -8,19 +8,25 @@ can now be expressed with units.
 
     import matplotlib.pyplot as plt
     import numpy as np
-    from matplotlib.dates import HourLocator, DateFormatter
+    from matplotlib.dates import HourLocator, ConciseDateFormatter
     from matplotlib.ticker import AutoMinorLocator
 
-    fig, ax = plt.subplots()
+    fig, ax = plt.subplots(layout='constrained')
     date_first = np.datetime64('2020-01-01', 'D')
     date_last = np.datetime64('2020-01-11', 'D')
 
     arr = [[i+j for i in range(10)] for j in range(10)]
 
     ax.imshow(arr, origin='lower', extent=[1, 11, date_first, date_last])
 
-    ax.yaxis.set_major_formatter(DateFormatter('%d/%m/%y:- %H00hours'))
-    ax.yaxis.set_major_locator(HourLocator(byhour=[0, 6, 12, 18, 24]))
+    locator = HourLocator(byhour=[0, 12, 24])
```

Comment:
Maybe add a quick comment like `# Set nicer tick label formatting` above this block to separate it from the plotting code above?